### PR TITLE
Opensearch Connector - Fix list table to exclude inaccessible indices

### PR DESCRIPTION
**Pull request by satheeshms**
## Description
We had a restrictive AWS role allowing users to query only certain indexes. We noticed that indexes that are not accessible are still listed down in Trino or Starburst, though they can not be queried as permission denied. We are planning a multitenant system and can't afford to show one tenant's index/table to other tenants.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/client/OpenSearchClient.java [hunk 1]
```diff
@@ -368,7 +371,8 @@ public boolean indexExists(String index)
             return response.getStatusLine().getStatusCode() == 200;
         }
         catch (ResponseException e) {
-            if (e.getResponse().getStatusLine().getStatusCode() == 404) {
+            if (e.getResponse().getStatusLine().getStatusCode() == 404 ||
+                    e.getResponse().getStatusLine().getStatusCode() == 403) {
```
Comment (ebyhr) [inline plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/client/OpenSearchClient.java:375]: Let's extract a constant as `Set<Integer>`.

File: plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/client/OpenSearchClient.java [hunk 2]
```diff
@@ -382,33 +386,35 @@ public List<String> getIndexes()
...
-                    if (docsCount == 0 && deletedDocsCount == 0) {
-                        try {
-                            // without documents, the index won't have any dynamic mappings, but maybe there are some explicit ones
-                            if (getIndexMetadata(index).schema().fields().isEmpty()) {
-                                continue;
-                            }
+                    indices.add(root.get(i).get("index").asText());
+                }
+
+                return indices.parallelStream().map(index -> {
```
Comment (ebyhr) [inline plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/client/OpenSearchClient.java:396]: We usually don't use `parallelStream` in this project. I would recommend focusing on the permission issue in this PR.

**Review by ebyhr — COMMENTED** [review]
Please fix commit author and use `rebase` instead of `merge`.

**Comment by satheeshms** [comment]
@ebyhr commits were messed with the wrong Github Id. Opened another one. Thanks for spending time reviewing the PR.
https://github.com/trinodb/trino/pull/22663

**Pull request closed without merging**